packages/oidc-dpop-client-browser/src/dpop/dpop.ts

```diff
@@ -30,14 +43,111 @@ import { JWK } from "node-jose";
  * @hidden
  */
 export async function generateJWK(
-  kty: "EC" | "OKP" | "RSA" | "oct",
+  kty: "EC" | "RSA",
   crvBitlength?: ECCurve | OKPCurve | number,
   parameters?: BasicParameters
 ): Promise<JSONWebKey> {
   const key = await JWK.createKey(kty, crvBitlength, parameters);
   return key.toJSON(true) as JSONWebKey;
 }
 
+/**
+ * Generates a Json Web Token (https://tools.ietf.org/html/rfc7519) containing
+ * the provided payload and using the signature algorithm specified in the options.
+ * @param payload The body of the JWT.
+ * @param key The key used for the signature.
+ * @param options
+ * @returns a 3-parts base64-encoded string, split by dots.
+ * @hidden
+ */
+export async function signJWT(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  payload: Record<string, any>,
+  key: JWKECKey | JWKOKPKey | JWKRSAKey | JWKOctKey,
+  options?: JoseJWT.SignOptions
+): Promise<string> {
+  const parsedKey = await JWK.asKey(key);
+  const convertedKey: string = parsedKey.toPEM(true);
+  const signed = JWT.sign(payload, convertedKey, {
+    ...(options as JWT.SignOptions),
+  });
```

Comment:
Also, even if you need to keep the assertion, is there a need to clone it?

```suggestion
  const signed = JWT.sign(payload, convertedKey, (options as JWT.SignOptions));
```

Reply:
Indeed 0d4acef4280dc4a09297631c98b42422fe10d335